dor/providers/process_basic_image.py

```diff
@@ -398,6 +402,147 @@ def run(self) -> None:
+            function=[UseFunction.service], format=UseFormat.image
+        )
+
+        try:
+            text_coordinates_result_file = self.accumulator.get_file(
+                function=[UseFunction.service], format=UseFormat.text_coordinates
+            )
+        except AccumulatorError:
+            text_coordinates_result_file = None
+
+        if text_coordinates_result_file:
+            alto_xml = text_coordinates_result_file.file_path.read_text()
+            alto_doc = AltoDoc.create(alto_xml)
+            plain_text = alto_doc.plain_text
+        else:
+            plain_text = ImageTextExtractor(image_path=service_result_file.file_path, language=self.language).text
+
+        text_file_path = self.accumulator.file_set_directory / file_info.path
+        text_file_path.write_text(plain_text)
+
+        try:
+            tech_metadata = TechnicalMetadata.create(text_file_path)
+        except JHOVEDocError:
+            return None
+
+        event = create_preservation_event(
+            "extract plain text with OCR", self.accumulator.collection_manager_email
+        )
+
+        self.accumulator.add_file(
+            ResultFile(
+                file_path=text_file_path,
+                tech_metadata=tech_metadata,
+                file_info=file_info,
+                source_file_result=service_result_file,
+                event=event,
+            )
+        )
+
+
+
+@dataclass
+class CreateTextAnnotations(Operation):
+
+    def run(self) -> None:
+        file_info = FileInfo(
+            identifier=self.accumulator.file_set_identifier.identifier,
+            basename=self.accumulator.file_set_identifier.basename,
+            uses=[UseFunction.service, UseFormat.text_annotations],
+            mimetype=Mimetype.JSON.value,
+        )
+
+        service_result_file = self.accumulator.get_file(
+            function=[UseFunction.service], format=UseFormat.image
+        )
+
+        text_coordinates_result_file = self.accumulator.get_file(
+            function=[UseFunction.service], format=UseFormat.text_coordinates
+        )
+
+        alto_xml = text_coordinates_result_file.file_path.read_text()
+        alto_doc = AltoDoc.create(alto_xml)
+        annotation_data = AnnotationData(alto_doc).data
+
+        annotation_file_path = self.accumulator.file_set_directory / file_info.path
+        annotation_file_path.write_text(json.dumps(annotation_data, indent=4))
+
+        try:
+            tech_metadata = TechnicalMetadata.create(annotation_file_path)
+        except JHOVEDocError:
+            return None
+
+        event = create_preservation_event(
+            "create annotation data from text coordinates", self.accumulator.collection_manager_email
+        )
+
+        self.accumulator.add_file(
+            ResultFile(
+                file_path=annotation_file_path,
+                tech_metadata=tech_metadata,
+                file_info=file_info,
+                source_file_result=service_result_file,
```

Comment:
I'm always specifying the source file as the Service JP2 that OCR was used on -- not the intermediate Alto XML. Hope that's correct.

Reply:
Fine for now. Hoping our talk with DigiPres adds more clarity to our events.